### DISCUSSION TITLE
Missing parameters in ozone bucket cli response

### DISCUSSION THREAD

**Discussion by vmahendrakarunravel** [apache/ozone#6578] [FAQ] [answered]
hello guys, 

I'm currently running ozone version 1.2.0 and was experimenting with creating buckets and such in my local instance. I noticed a missing parameter called owner in the response that I get when I run ozone sh bucket info. Initially, when I create a bucket and run the info I get the below response
ozone sh bucket info vol2/bucket11
```
{
  "metadata" : { },
  "volumeName" : "vol2",
  "name" : "bucket11",
  "storageType" : "DISK",
  "versioning" : false,
  "usedBytes" : 0,
  "usedNamespace" : 0,
  "creationTime" : "2024-04-23T13:37:23.733Z",
  "modificationTime" : "2024-04-23T13:37:23.733Z",
  "quotaInBytes" : -1,
  "quotaInNamespace" : -1,
  "bucketLayout" : "LEGACY",
  "owner" : "user",
  "link" : false
}
```
here we find the owner in the response but as soon as I set some space quota to the bucket the owner param goes missing.
```
{
  "metadata" : { },
  "volumeName" : "vol2",
  "name" : "bucket11",
  "storageType" : "DISK",
  "versioning" : false,
  "usedBytes" : 0,
  "usedNamespace" : 0,
  "creationTime" : "2024-04-23T13:37:23.733Z",
  "modificationTime" : "2024-04-23T13:39:39.443Z",
  "quotaInBytes" : 1024,
  "quotaInNamespace" : -1,
  "bucketLayout" : "LEGACY",
  "replicationConfig" : {
    "replicationFactor" : "ONE",
    "requiredNodes" : 1,
    "replicationType" : "RATIS"
  },
  "link" : false
}
```

can someone pls help me understand why this happening? 
also, buckets inherit the owner from its volume right? is it a similar case for keys as well?

**Comment by errose28** [comment, marked as answer]
> can someone pls help me understand why this happening?

You are hitting [HDDS-9110](https://issues.apache.org/jira/browse/HDDS-9110). I would definitely recommend upgrading to the latest version of Ozone: 1.4.0, and then 1.4.1 when it comes out. The amount of bugs fixed in the two years since 1.2.0 was released is quite large. This includes many fixes to quotas as well.

> buckets inherit the owner from its volume right?

I'm actually not sure, but I believe the bucket owner defaults to the user who creates the bucket.

> is it a similar case for keys as well?

Ozone currently does not have a concept of key ownership. There is a proposal to support this with Native ACLs in #4188 but no timeline for completion of the feature. As you can see it has been open for a while.
